src/rust/lib/graph-editor/src/component/node.rs

```diff
@@ -214,6 +220,11 @@ impl Node {
             selection_ref.set_target_position(0.0);
         });
 
+        // TODO this is sample functionality. Needs to be replaced with logic creating ports.
+        self.data.port_manager.set_parent(self.downgrade());
```

Comment:
The `PortManager` contains logic to create new ports. To be able to to that it needs to know who is supposed to be the parent of those nodes to call `parent.add_child`.

I had assumed that keeping around a `WeakNode` for that purpose would be unproblematic. It would only get upgraded when trying to add a child. Is there an issue with that?